homeassistant/components/home_connect/binary_sensor.py

```diff
@@ -125,59 +126,41 @@ def available(self) -> bool:
 
     async def async_update(self) -> None:
         """Update the binary sensor's status."""
-        state = self.device.appliance.status.get(self._update_key, {})
-        if not state:
+        if not self.device.appliance.status or not (
+            status := self.device.appliance.status.get(self.bsh_key, {}).get(ATTR_VALUE)
+        ):
             self._attr_is_on = None
             return
-
-        value = state.get(ATTR_VALUE)
-        if self._false_value_list and self._true_value_list:
-            if value in self._false_value_list:
-                self._attr_is_on = False
-            elif value in self._true_value_list:
-                self._attr_is_on = True
-            else:
-                _LOGGER.warning(
-                    "Unexpected value for HomeConnect %s state: %s", self._type, state
-                )
-                self._attr_is_on = None
-        elif isinstance(value, bool):
-            self._attr_is_on = value
-        else:
-            _LOGGER.warning(
-                "Unexpected value for HomeConnect %s state: %s", self._type, state
-            )
+        if self.entity_description.boolean_map:
+            self._attr_is_on = self.entity_description.boolean_map.get(status)
+        elif status not in [True, False]:
             self._attr_is_on = None
+        else:
+            self._attr_is_on = status
         _LOGGER.debug("Updated, new state: %s", self._attr_is_on)
 
 
-class HomeConnectFridgeDoorBinarySensor(HomeConnectEntity, BinarySensorEntity):
-    """Binary sensor for Home Connect Fridge Doors."""
+class HomeConnectDoorBinarySensor(HomeConnectBinarySensor):
+    """Binary sensor for Home Connect Generic Door."""
 
-    entity_description: HomeConnectBinarySensorEntityDescription
+    _attr_has_entity_name = False
```

Comment:
Why?

Reply:
Because of #126158 simple door binary sensor will be deprecated, so I thought that translation wasn't necessary